tools/cpp/generate_compilation_database.sh

```diff
@@ -1,25 +1,29 @@
 #!/bin/bash
 
 # Generates a compile_commands.json file at $(bazel info execution_root) for
-# your Clang tooling needs.
+# the given file path.
 
 set -e
 
+FILENAME=${1:?Missing required source path}
 bazel build \
   --experimental_action_listener=//kythe/cxx/tools/generate_compile_commands:extract_json \
   --noshow_progress \
   --noshow_loading_progress \
-  $(bazel query 'kind(cc_.*, //...) - attr(tags, manual, //...)') > /dev/null
+  --output_groups=compilation_outputs \
+  --compile_one_dependency \
+  "$FILENAME" > /dev/null
 
 BAZEL_ROOT="$(bazel info execution_root)"
 pushd "$BAZEL_ROOT" > /dev/null
-echo "[" > compile_commands.json
-COUNT=0
 find . -name '*.compile_command.json' -print0 | while read -r -d '' fname; do
-  if ((COUNT++)); then
-    echo ',' >> compile_commands.json
-  fi
   sed -e "s|@BAZEL_ROOT@|$BAZEL_ROOT|g" < "$fname" >> compile_commands.json
+  echo "" >> compile_commands.json
 done
-echo "]" >> compile_commands.json
+# Decompose, insert and keep the most recent entry for a given file, then
+# recombine.
+sed 's/\(^[[]\)\|\([],]$\)//;/^$/d;' < compile_commands.json \
```

Comment:
I'll take your word for it. If we need to update this though, I'd be in favor of writing a custom binary so it is a little easier to read.

Reply:
It would be much simpler if JSON allowed trailing commas on the last element in a list, but noooooo.

Reply:
FWIW, the 'jq' tool: https://stedolan.github.io/jq/
is really great and I highly recommend it.